### PR TITLE
Windows - Install-Msys2.ps1 - minor reliability, taskkill, logging

### PR DESCRIPTION
**Pull request by MSP-Greg**
# Description
New tool, Bug fixing, or Improvement?  **Improvement**

1. MSYS2 - add small changes to make MSYS2 installation/update via script more stable.

2. Update logging code for both install/update and final package listing.

Tested both locally and with GHA

### CODE REVIEW

File: images/win/scripts/Installers/Install-Msys2.ps1 [hunk 1]
```diff
@@ -17,7 +19,7 @@ $msys2Uri = ((Invoke-RestMethod $msys2_release).assets | Where-Object {
 
 $msys2File = "$env:TEMP\msys2.tar.xz"
 
-# Download the latest msys2 x86_64
+# Download the latest msys2 x86_64, filename includes release date
 Write-Host "Starting msys2 download using $($msys2Uri.split('/')[-1])"
 (New-Object System.Net.WebClient).DownloadFile($msys2Uri, $msys2File)
```
Comment (maxim-lobanov) [inline images/win/scripts/Installers/Install-Msys2.ps1:24]: I suggest to use helper `Start-DownloadWithRetry` (https://github.com/actions/virtual-environments/blob/master/images/win/scripts/ImageHelpers/InstallHelpers.ps1#L218) here

Comment (MSP-Greg) [reply images/win/scripts/Installers/Install-Msys2.ps1]: @maxim-lobanov  Good idea.  Done.

**Comment by MSP-Greg** [comment]
JFYI, Windows 2019 has C:\Windows\System32\tar.exe, Windows 2016 does not.  When 2016 is dropped, code can be revised to remove dependency on Git tar.  Seems messy to include conditional code for it now...

**Comment by MSP-Greg** [comment]
Ok, I'm confused.  Azure Pipelines crashed and burned.

I just ran the script in Actions (copied the ImageHelpers folder also) and it installed fine:
https://github.com/MSP-Greg/actions-msys2-pkgs/runs/707467166?check_suite_focus=true#step:3:610

Happy to change whatever is needed...

**Comment by vsafonkin** [comment]
Hi @MSP-Greg , the reason for the fails is this line:
```
$dash = "—" * 40
```
This dash is incorrect symbol.

**Comment by MSP-Greg** [comment]
Apologies, encoding seemed suspect, but I didn't think Actions and Pipelines would using different encodings.  I recall wrestling with it on AppVeyor...

Fixed with a standard dash.

**Comment by MSP-Greg** [comment]
It appears that the last CI failed before it got to Install-Msys2.ps1 ?

**Comment by miketimofeev** [comment]
@MSP-Greg yeah, we're facing this issue at the moment
https://github.com/actions/virtual-environments/issues/940

**Comment by MSP-Greg** [comment]
I pushed a change, adding the ~ folder (`C:\msys64\home\$env:USERNAME`).  Kept forgetting about it, sorry...

**Comment by al-cheb** [comment]
> I pushed a change, adding the ~ folder (`C:\msys64\home\$env:USERNAME`). Kept forgetting about it, sorry...

Please revert it. We can't preserver user name for different environments.

**Comment by MSP-Greg** [comment]
@al-cheb

> Please revert it

Please reconsider.

1. It's only a folder

2. MSYS2's normal shell sets it as the default for `~`.  Note that Git's shell uses `/c/Users/$env:USERNAME`.

3. I suspect many of the installed apps write to HKCU, `runneradmin` appears in a few ENV variables, and appears twice in `Path`.  Hence, `USERNAME` is kind of baked in anyway.

4. Although most app's should check for a folder's existence, test scripts may not do so.

5. This may be helpful when using cross-platform bash in CI.

6. I'm the owner of [MSP-Greg/setup-ruby-pkgs](https://github.com/MSP-Greg/setup-ruby-pkgs), and contributed to [ruby/setup-ruby](https://github.com/ruby/setup-ruby).  In setup-ruby-pkgs, I've got code that adds this (`~`) folder.  I didn't list a reason for doing so, but it must have been due to an issue, probably related to item 4.  It really should be part of the install.

If you're adamant, I'll remove.  Thanks.

**Comment by al-cheb** [comment]
@MSP-Greg, After completion image generation all users are delete from image.

**Comment by miketimofeev** [comment]
@MSP-Greg as @al-cheb said this addition doesn't make sense, unfortunately. This runneradmin user is being added during VM creation only.

**Comment by MSP-Greg** [comment]
@al-cheb

If the user name for building the image is 'runneradmin'?  Regardless, removed the code.

**Comment by miketimofeev** [comment]
@MSP-Greg "packer" user is being used as you can see from the template
https://github.com/actions/virtual-environments/blob/2b6b0382e8a42fefaf9510a51e7708230c4c6ab3/images/win/Windows2019-Azure.json#L30

File: images/win/scripts/Installers/Install-Msys2.ps1 [hunk 2]
```diff
@@ -31,63 +31,60 @@ $env:PATH = "$gitPath\usr\bin;$gitPath\mingw64\bin;$origPath"
...
-pacman -Su --noconfirm
-
-# Force stop gpg-agent to continue installation
-Get-Process gpg-agent -ErrorAction SilentlyContinue | Stop-Process -Force
-
-Write-Host "pacman --noconfirm -Syyuu"
+Write-Host "`n$dash pacman --noconfirm -Syyuu"
 pacman.exe -Syyuu --noconfirm
-pacman.exe -Syuu --noconfirm
+taskkill /f /fi "MODULES eq msys-2.0.dll"
```
[the remark this answers is not in the document]
Comment (eine) [reply images/win/scripts/Installers/Install-Msys2.ps1:50]: @alepauly, see https://github.com/actions/virtual-environments/pull/906#issuecomment-632419897. The procedure in this PR is different from the recommended in those references (`-Syyuu` vs `-Syuu`, and `-Syuu` vs `-Suu`). However, I think that this first execution of `taskkill /f /fi "MODULES eq msys-2.0.dll"` is required. I'm not sure about others, tho.

[the remark this answers is not in the document]
Comment (MSP-Greg) [reply images/win/scripts/Installers/Install-Msys2.ps1:50]: I thought I'd err on the cautious side, as the script should be bullet-proof.  I intend to keep an eye on future CI logs to see if all the taskkill statements are needed.  Some were included based on doing the install in Actions (mult vm's on one system), which may not be the case for the Pipeliines runs...

**Comment by eine** [comment]
Ref #355

File: images/win/scripts/Installers/Install-Msys2.ps1 (hunk 2 above)
[the remark this answers is not in the document]
Comment (eine) [reply images/win/scripts/Installers/Install-Msys2.ps1:50]: @MSP-Greg, I think that the first `y` is irrelevant (i.e. `Syyuu` or `Syuu`). However, I believe that using `y` in following statements is what might make `taskkill` to be required again. Anyway, this should be fixed upstream, albeit "experimental" (https://github.com/msys2/msys2-installer/releases/tag/2020-05-22). We should not need it in the near future.

[the remark this answers is not in the document]
Comment (MSP-Greg) [reply images/win/scripts/Installers/Install-Msys2.ps1:50]: @eine

> Anyway, this should be fixed upstream

As mentioned above, 'the script should be bullet-proof'.  If an extra few seconds are added to the install time to do that, so be it.  As MSYS2 is tested better re scripted install, etc, the install script may be able to be simplified.

Both Ruby itself and several extension gems are running CI using the new MSYS2 install, and all are passing with no changes.  All are using the gcc tools, the bash shell is used less frequently.

[the remark this answers is not in the document]
Comment (eine) [reply images/win/scripts/Installers/Install-Msys2.ps1:50]: > Both Ruby itself and several extension gems are running CI using the new MSYS2 install, and all are passing with no changes. All are using the gcc tools, the bash shell is used less frequently.

In fact, most users of [eine/setup-msys2](https://github.com/eine/setup-msys2) are using `update: true`. Hence, apart from the Ruby extensions gems, there is intensive testing going on. See eine/setup-msys2#20. For example in [ghdl/ghdl:actions](https://github.com/ghdl/ghdl/actions) nightly packages for MSYS2 are being built with setup-msys2.

**Pull request merged**